### PR TITLE
feat(LIB-1340): Handle singlefile in FzUpload

### PR DESCRIPTION
**Pull request by Crish15**
Jira: [LIB-1340](https://fiscozen.atlassian.net/browse/LIB-1340)

### CODE REVIEW

File: packages/upload/src/FzUpload.vue [hunk 1]
```diff
@@ -68,12 +68,18 @@ const props = withDefaults(defineProps<FzUploadProps>(), {
...
+const emit = defineEmits(["fzupload:change"]);
 const model = defineModel<File[]>();
 
 const input = ref<HTMLInputElement | null>(null);
 const urlByFileMap = ref(new Map<File, string>());
 
+onMounted(() => {
+  if(model.value && model.value.length > 0 && !props.multiple) {
+    console.warn("[Fiscozen Design System]: FzUpload prop 'multiple' is set to false, but multiple files are provided. Only the first file will be used.");
+  }
```
Comment (oxcened) [inline packages/upload/src/FzUpload.vue:80]: I don't see any logic that stops me from passing more than one file to the model

File: packages/upload/src/FzUpload.vue [hunk 2]
```diff
@@ -93,11 +99,16 @@ function handleDrop(event: DragEvent) {
 function handleInputChange(event: Event) {
   if (!input.value?.files) return;
   addFiles([...input.value.files]);
+  emit("fzupload:change", model.value);
```
Comment (oxcened) [inline packages/upload/src/FzUpload.vue:102]: What is this for? I don't think it does what you want, you are emitting the old model

File: packages/upload/src/FzUpload.vue [hunk 3]
```diff
@@ -93,11 +99,16 @@ function handleDrop(event: DragEvent) {
...
   input.value.value = "";
 }
 
 function addFiles(filesToAdd: File[]) {
-  model.value = [...(model.value ?? []), ...filesToAdd];
+  if(props.multiple) {
```
Comment (oxcened) [inline packages/upload/src/FzUpload.vue:107]: there's some spaces missing here and there, you should run prettier

File: packages/upload/src/FzUpload.vue [hunk 4]
```diff
@@ -93,11 +99,16 @@ function handleDrop(event: DragEvent) {
...
+    model.value = [...(model.value ?? []), ...filesToAdd];
+  } else {
+    model.value = [...filesToAdd];
```
Comment (oxcened) [inline packages/upload/src/FzUpload.vue:110]: It's not granted that you only have one file. If it comes from the input file then yes it should. But if it's come drag and drop, you have to handle that yourself.

File: packages/upload/src/FzUpload.vue (hunk 2 above)
Comment (Crish15) [reply packages/upload/src/FzUpload.vue:102]: addFiles assign the new model

File: packages/upload/src/FzUpload.vue [hunk 5]
```diff
@@ -68,12 +68,33 @@ const props = withDefaults(defineProps<FzUploadProps>(), {
...
+const emit = defineEmits(["fzupload:change"]);
+const model = defineModel<File[]>({
+  default: [],
+});
 
 const input = ref<HTMLInputElement | null>(null);
 const urlByFileMap = ref(new Map<File, string>());
 
+onMounted(() => {
+  if (model.value && model.value.length > 0 && !props.multiple) {
```
Comment (oxcened) [inline packages/upload/src/FzUpload.vue:80]:
```suggestion
  if (model.value?.length > 1 && !props.multiple) {
```

File: packages/upload/src/FzUpload.vue [hunk 6]
```diff
@@ -68,12 +68,33 @@ const props = withDefaults(defineProps<FzUploadProps>(), {
...
+  }
+});
+
+watch(
+  () => model.value,
+  (newFiles: File[]) => {
+    if (newFiles.length > 1 && !props.multiple) {
+      model.value = [newFiles[0]];
+    }
+    emit("fzupload:change", model.value);
```
Comment (oxcened) [inline packages/upload/src/FzUpload.vue:94]: I think this is redundant, there's already update:modelValue which does the exact same thing

File: packages/upload/src/FzUpload.vue [hunk 7]
```diff
@@ -68,12 +68,33 @@ const props = withDefaults(defineProps<FzUploadProps>(), {
...
+    model.value = [model.value[0]];
+  }
+});
+
+watch(
+  () => model.value,
+  (newFiles: File[]) => {
+    if (newFiles.length > 1 && !props.multiple) {
+      model.value = [newFiles[0]];
+    }
```
Comment (oxcened) [inline packages/upload/src/FzUpload.vue:93]: I would do this in the setter of the model

File: packages/upload/src/FzUpload.vue [hunk 8]
```diff
@@ -68,12 +68,33 @@ const props = withDefaults(defineProps<FzUploadProps>(), {
...
+});
 
 const input = ref<HTMLInputElement | null>(null);
 const urlByFileMap = ref(new Map<File, string>());
 
+onMounted(() => {
+  if (model.value && model.value.length > 0 && !props.multiple) {
+    console.warn(
+      "[Fiscozen Design System]: FzUpload prop 'multiple' is set to false, but multiple files are provided. Only the first file will be used.",
+    );
```
Comment (oxcened) [inline packages/upload/src/FzUpload.vue:83]: this warning could also go in the setter, even more so considering you're only checking at mount and never after that

File: packages/upload/src/FzUpload.vue (hunk 6 above)
Comment (Crish15) [reply packages/upload/src/FzUpload.vue:94]: emitting the change allows the component to be used without passing a model. It is also easier to understand than ‘update:modelValue’.

File: packages/upload/src/FzUpload.vue [hunk 9]
```diff
@@ -68,12 +68,31 @@ const props = withDefaults(defineProps<FzUploadProps>(), {
   buttonLabel: "Carica",
   dragAndDropLabel: "o trascina qui",
 });
-const emit = defineEmits([]);
-const model = defineModel<File[]>();
+const emit = defineEmits(["fzupload:change"]);
```
Comment (oxcened) [inline packages/upload/src/FzUpload.vue:71]:
```suggestion
const emit = defineEmits<{
  fzupload:change: [files: File[]]
}>();
```

File: packages/upload/src/FzUpload.vue [hunk 10]
```diff
@@ -97,7 +116,13 @@ function handleInputChange(event: Event) {
...
 
 function addFiles(filesToAdd: File[]) {
-  model.value = [...(model.value ?? []), ...filesToAdd];
+  if (props.multiple) {
+    model.value = [...(model.value ?? []), ...filesToAdd];
+  } else {
+    model.value = [filesToAdd[0]];
+  }
+
+  emit("fzupload:change", model.value);
```
Comment (oxcened) [inline packages/upload/src/FzUpload.vue:125]: this works as is? I thought changing a model was async

Comment (Crish15) [reply packages/upload/src/FzUpload.vue:125]: it should works, but to be safe I changed it

**Review by oxcened — APPROVED** [review]
lgtm

**Pull request merged**